src/unix/apple/groups.rs

```diff
@@ -0,0 +1,124 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{common::Gid, Group, GroupInner};
+
+use libc::{endgrent, getgrent, setgrent};
+use std::collections::HashMap;
+
+pub(crate) fn get_groups(groups: &mut Vec<Group>) {
+    groups.clear();
+
+    let mut groups_map = HashMap::with_capacity(10);
```

Comment:
Instead of matching on group name, wouldn't it be better to match on `gid` instead?

Reply:
Some Unix platforms allow creating groups with the same GID but different names (like `groupadd -o` on GNU/Linux). How should them be handled?

Reply:
Oh I see. Just double-checked and you're right, the group name is unique.